src/tests/setup.ts

```diff
@@ -0,0 +1,60 @@
+import { Centrifuge } from '../Centrifuge.js'
+import { TenderlyFork } from './tenderly.js'
+import { sepolia } from 'viem/chains'
+
+declare global {
+  namespace Mocha {
+    interface Context {
+      context: TestContext & TestContext
+    }
+  }
+}
+
+class TestContext {
+  public centrifuge!: Centrifuge
+  public tenderlyFork!: TenderlyFork
+  public allTestsSucceeded = true
+
+  async initialize() {
+    this.tenderlyFork = await TenderlyFork.create(sepolia)
+    this.centrifuge = new Centrifuge({
+      environment: 'demo',
+      rpcUrls: {
+        11155111: this.tenderlyFork.rpcUrl,
+      },
+    })
+    this.centrifuge.setSigner(this.tenderlyFork.account)
+  }
+
+  async cleanup() {
+    if (this.tenderlyFork && this.allTestsSucceeded) {
+      await this.tenderlyFork.deleteTenderlyRpcEndpoint()
+    }
+    console.log('A test has failed, RPC endpoint will not be deleted', this.tenderlyFork.rpcUrl)
+  }
+}
+
+export function withContext(fn: (this: Mocha.Context & { context: TestContext }) => Promise<void>) {
+  return function (this: Mocha.Context) {
+    return fn.call(this)
+  }
+}
+
+const testContext = new TestContext()
```

Comment:
can't we simply export `context`? Then we don't have to wrap every test in a `withContext`

Reply:
yeah that's definitely cleaner!